### ISSUE TITLE
Impossible to import geopandas (libjson problem)

### ISSUE THREAD

**Issue by aborruso** [geopandas/geopandas#727]
Hi,
I have Ubuntu 18 and I have a `libjson` dependency problem when I try to use geopandas. How to solve it?

Thank you

```python
!conda install geopandas -c conda-forge
```

    Fetching package metadata .............
    Solving package specifications: .
    
    # All requested packages already installed.
    # packages in environment at /srv/conda:
    #
    geopandas                 0.3.0                    py36_0    conda-forge

```python
import geopandas as gpd
```

    ---------------------------------------------------------------------------

    ImportError                               Traceback (most recent call last)

    <ipython-input-16-a62d01c1d62e> in <module>()
    ----> 1 import geopandas as gpd
    

    /srv/conda/lib/python3.6/site-packages/geopandas/__init__.py in <module>()
          2 from geopandas.geodataframe import GeoDataFrame
          3 
    ----> 4 from geopandas.io.file import read_file
          5 from geopandas.io.sql import read_postgis
          6 from geopandas.tools import sjoin

    /srv/conda/lib/python3.6/site-packages/geopandas/io/file.py in <module>()
          1 import os
          2 
    ----> 3 import fiona
          4 import numpy as np
          5 

    /srv/conda/lib/python3.6/site-packages/fiona/__init__.py in <module>()
         67 from six import string_types
         68 
    ---> 69 from fiona.collection import Collection, BytesCollection, vsi_path
         70 from fiona._drivers import driver_count, GDALEnv
         71 from fiona.drvsupport import supported_drivers

    /srv/conda/lib/python3.6/site-packages/fiona/collection.py in <module>()
          7 
          8 from fiona import compat
    ----> 9 from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
         10 from fiona.ogrext import Session, WritingSession
         11 from fiona.ogrext import (

    ImportError: libjson-c.so.2: cannot open shared object file: No such file or directory

**Comment by jorisvandenbossche** [comment]
Can you show the full output of 'conda list' ?

**Comment by aborruso** [comment]
Hi @jorisvandenbossche this one. Thank you

```
# packages in environment at /srv/conda:
#
affine                    2.2.0                     <pip>
agate                     1.6.1                     <pip>
agate-dbf                 0.2.0                     <pip>
agate-excel               0.2.2                     <pip>
agate-sql                 0.5.3                     <pip>
altair                    2.0.1                      py_0    conda-forge
asn1crypto                0.22.0           py36h265ca7c_1    defaults
attrs                     18.1.0                    <pip>
Babel                     2.5.3                     <pip>
backcall                  0.1.0                      py_0    conda-forge
bash_kernel               0.7.1                     <pip>
beautifulsoup4            4.6.0                    py36_0    conda-forge
bleach                    2.1.3                      py_0    conda-forge
boost                     1.66.0                   py36_1    conda-forge
boost-cpp                 1.66.0                        1    conda-forge
bs4                       0.0.1                     <pip>
bzip2                     1.0.6                         1    conda-forge
ca-certificates           2018.4.16                     0    conda-forge
cairo                     1.14.10                       0    conda-forge
Cartopy                   0.16.0                    <pip>
cchardet                  1.1.3                     <pip>
certifi                   2018.4.16                py36_0    conda-forge
cffi                      1.10.0           py36had8d393_1    defaults
chardet                   3.0.4            py36h0f667ec_1    defaults
click                     6.7                        py_1    conda-forge
click-default-group       1.2                       <pip>
click-plugins             1.0.3                    py36_0    conda-forge
cligj                     0.4.0                    py36_0    conda-forge
conda                     4.3.30           py36h5d9f9f4_0    defaults
conda-env                 2.6.0                h36134e3_1    defaults
contextily                0.99.0                    <pip>
cryptography              2.0.3            py36ha225213_1    defaults
cssselect                 1.0.3                     <pip>
csvkit                    1.0.3                     <pip>
csvs-to-sqlite            0.8.1                     <pip>
curl                      7.59.0                        1    conda-forge
cycler                    0.10.0                   py36_0    conda-forge
cython                    0.28.2                   py36_0    conda-forge
datapackage               1.2.6                     <pip>
dateparser                0.7.0                     <pip>
dbfread                   2.0.7                     <pip>
dbus                      1.11.0                        0    conda-forge
decorator                 4.3.0                      py_0    conda-forge
descartes                 1.1.0                      py_1    conda-forge
docutils                  0.14                      <pip>
entrypoints               0.2.3                    py36_1    conda-forge
et-xmlfile                1.0.1                     <pip>
expat                     2.2.5                         0    conda-forge
fake-useragent            0.1.10                    <pip>
fiona                     1.7.11                   py36_3    conda-forge
fontconfig                2.12.6                        0    conda-forge
freetype                  2.8.1                         0    conda-forge
freexl                    1.0.5                         0    conda-forge
future                    0.16.0                    <pip>
gdal                      2.2.4                    py36_0    conda-forge
geographiclib             1.49                      <pip>
geopandas                 0.3.0                    py36_0    conda-forge
geopy                     1.14.0                    <pip>
geos                      3.6.2                         1    conda-forge
geotiff                   1.4.2                         1    conda-forge
gettext                   0.19.8.1                      0    conda-forge
giflib                    5.1.4                         0    conda-forge
glib                      2.55.0                        0    conda-forge
gmp                       6.1.2                         0    conda-forge
goodtables                2.0.2                     <pip>
gst-plugins-base          1.8.0                         0    conda-forge
gstreamer                 1.8.0                         1    conda-forge
hdf4                      4.2.13                        0    conda-forge
hdf5                      1.10.1                        2    conda-forge
html5lib                  1.0.1                      py_0    conda-forge
httpie                    0.9.9                     <pip>
icu                       58.2                          0    conda-forge
idna                      2.6              py36h82fb2a8_1    defaults
ijson                     2.3                       <pip>
intel-openmp              2018.0.0                      8    defaults
ipykernel                 4.8.2                    py36_0    conda-forge
ipython                   6.3.1                    py36_0    conda-forge
ipython-sql               0.3.9                     <pip>
ipython_genutils          0.2.0                    py36_0    conda-forge
ipywidgets                7.1.1                    py36_0    conda-forge
isodate                   0.6.0                     <pip>
jdcal                     1.4                       <pip>
jedi                      0.12.0                   py36_0    conda-forge
jinja2                    2.10                     py36_0    conda-forge
jpeg                      9b                            2    conda-forge
json-c                    0.13.1               h470a237_0    conda-forge
jsonlines                 1.2.0                     <pip>
jsonpointer               1.14                      <pip>
jsonschema                2.6.0                    py36_1    conda-forge
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                    py36_0    conda-forge
jupyter_console           5.2.0                    py36_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterlab                0.31.5                   py36_1    conda-forge
jupyterlab_launcher       0.10.5                   py36_0    conda-forge
kealib                    1.4.7                         4    conda-forge
kiwisolver                1.0.1                    py36_1    conda-forge
krb5                      1.14.6                        0    conda-forge
leather                   0.3.3                     <pip>
libdap4                   3.18.3                        2    conda-forge
libedit                   3.1.20170329                  0    conda-forge
libffi                    3.2.1                h4deb6c0_3    defaults
libgcc                    7.2.0                h69d50b8_2    defaults
libgcc-ng                 7.2.0                h7cc24e2_2    defaults
libgdal                   2.2.4                         2    conda-forge
libgfortran               3.0.0                         1    defaults
libgfortran-ng            7.2.0                hdf63c60_3    defaults
libiconv                  1.15                          0    conda-forge
libkml                    1.3.0                         6    conda-forge
libnetcdf                 4.6.1                         2    conda-forge
libpng                    1.6.34                        0    conda-forge
libpq                     9.6.3                         0    conda-forge
libsodium                 1.0.16                        0    conda-forge
libspatialindex           1.8.5                         1    conda-forge
libspatialite             4.3.0a                       19    conda-forge
libssh2                   1.8.0                         2    conda-forge
libstdcxx-ng              7.2.0                h7a57d05_2    defaults
libtiff                   4.0.9                         0    conda-forge
libxcb                    1.13                          0    conda-forge
libxml2                   2.9.8                         0    conda-forge
linear-tsv                1.1.0                     <pip>
lml                       0.0.1                     <pip>
lxml                      4.2.1                     <pip>
markupsafe                1.0                      py36_0    conda-forge
matplotlib                2.2.2                    py36_1    conda-forge
mercantile                1.0.2                     <pip>
mistune                   0.8.3                    py36_1    conda-forge
mkl                       2018.0.2                      1    defaults
mkl_fft                   1.0.2                    py36_0    conda-forge
mkl_random                1.0.1                    py36_0    conda-forge
more-itertools            4.1.0                     <pip>
munch                     2.3.2                      py_0    conda-forge
nbconvert                 5.3.1                      py_1    conda-forge
nbformat                  4.4.0                    py36_0    conda-forge
ncurses                   5.9                          10    conda-forge
notebook                  5.5.0                    py36_0    conda-forge
nteract-on-jupyter        1.7.0                     <pip>
numpy                     1.14.2           py36hdbf6ddf_1    defaults
openjpeg                  2.3.0                         2    conda-forge
openpyxl                  2.5.3                     <pip>
openssl                   1.0.2o                        0    conda-forge
pandas                    0.23.0                   py36_0    conda-forge
pandas                    0.20.3                    <pip>
pandoc                    2.2                           0    conda-forge
pandocfilters             1.4.2                    py36_0    conda-forge
parse                     1.8.2                     <pip>
parsedatetime             2.4                       <pip>
parso                     0.2.0                      py_0    conda-forge
pcre                      8.41                          1    conda-forge
pexpect                   4.5.0                    py36_0    conda-forge
pickleshare               0.7.4                    py36_0    conda-forge
Pillow                    5.1.0                     <pip>
pip                       9.0.3                    py36_0    conda-forge
pixman                    0.34.0                        2    conda-forge
pluggy                    0.6.0                     <pip>
poppler                   0.61.1                        3    conda-forge
poppler-data              0.4.9                         0    conda-forge
prettytable               0.7.2                     <pip>
proj4                     4.9.3                         5    conda-forge
prompt_toolkit            1.0.15                   py36_0    conda-forge
psycopg2                  2.7.4                    py36_0    conda-forge
ptyprocess                0.5.2                    py36_0    conda-forge
py                        1.5.3                     <pip>
py-lru-cache              0.1.4                     <pip>
pycosat                   0.6.2            py36h1a0ea17_1    defaults
pycparser                 2.18             py36hf9f622e_1    defaults
pyee                      5.0.0                     <pip>
pyexcel                   0.5.8                     <pip>
pyexcel-cli               0.0.3                     <pip>
pyexcel-io                0.5.7                     <pip>
pygments                  2.2.0                    py36_0    conda-forge
pyopenssl                 17.2.0           py36h5cc804b_0    defaults
pyparsing                 2.2.0                    py36_0    conda-forge
pyppeteer                 0.0.17                    <pip>
pyproj                    1.9.5.1                  py36_0    conda-forge
pyqt                      5.6.0                    py36_5    conda-forge
pyquery                   1.4.0                     <pip>
pysal                     1.14.3                   py36_0    conda-forge
pyshp                     1.2.12                    <pip>
pysocks                   1.6.7            py36hd97a5b1_1    defaults
pytest                    3.5.1                     <pip>
python                    3.6.5                         1    conda-forge
python-dateutil           2.7.2                      py_0    conda-forge
python-slugify            1.2.5                     <pip>
pytimeparse               1.1.7                     <pip>
pytz                      2018.4                     py_0    conda-forge
PyYAML                    3.12                      <pip>
pyzmq                     17.0.0                   py36_4    conda-forge
qt                        5.6.2                         7    conda-forge
qtconsole                 4.3.1                    py36_0    conda-forge
rasterio                  0.36.0                    <pip>
readline                  7.0                           0    conda-forge
regex                     2018.2.21                 <pip>
requests                  2.18.4           py36he2e5f8d_1    defaults
requests-html             0.9.0                     <pip>
rfc3986                   1.1.0                     <pip>
rtree                     0.8.3                    py36_0    conda-forge
ruamel_yaml               0.11.14          py36ha2fb22d_2    defaults
scipy                     1.1.0            py36hfc37229_0    defaults
send2trash                1.5.0                      py_0    conda-forge
setuptools                39.1.0                   py36_0    conda-forge
shapely                   1.6.4                    py36_0    conda-forge
simpleeval                0.9.5                     <pip>
simplegeneric             0.8.1                    py36_0    conda-forge
sip                       4.18                     py36_1    conda-forge
six                       1.11.0                   py36_1    conda-forge
snuggs                    1.4.1                     <pip>
sqlalchemy                1.2.7            py36h65ede16_0    conda-forge
sqlite                    3.20.1                        2    conda-forge
sqlparse                  0.2.4                     <pip>
statistics                1.0.3.5                   <pip>
tableschema               1.0.13                    <pip>
tabulator                 1.14.0                    <pip>
terminado                 0.8.1                    py36_0    conda-forge
testpath                  0.3.1                    py36_0    conda-forge
texttable                 1.2.1                     <pip>
tk                        8.6.7                         0    conda-forge
toolz                     0.9.0                      py_0    conda-forge
tornado                   4.5.3                    py36_0    conda-forge
traitlets                 4.3.2                    py36_0    conda-forge
typing                    3.6.4                    py36_0    conda-forge
tzlocal                   1.5.1                     <pip>
unicodecsv                0.14.1                    <pip>
Unidecode                 1.0.22                    <pip>
urllib3                   1.22             py36hbe7ace6_0    defaults
util-linux                2.21                          0    defaults
vega3                     0.13.0                   py36_0    conda-forge
vega_datasets             0.5.0                      py_0    conda-forge
w3lib                     1.19.0                    <pip>
wcwidth                   0.1.7                    py36_0    conda-forge
webencodings              0.5                      py36_0    conda-forge
websockets                4.0.1                     <pip>
wheel                     0.31.0                   py36_0    conda-forge
widgetsnbextension        3.2.1                    py36_0    conda-forge
xerces-c                  3.2.0                         0    conda-forge
xlrd                      1.1.0                     <pip>
xmltodict                 0.11.0                    <pip>
xorg-libxau               1.0.8                         3    conda-forge
xorg-libxdmcp             1.1.2                         3    conda-forge
xz                        5.2.3                         0    conda-forge
yaml                      0.1.7                h96e3832_1    defaults
yq                        2.6.0                     <pip>
zeromq                    4.2.5                         1    conda-forge
zlib                      1.2.11                        0    conda-forge
```

**Comment by jorisvandenbossche** [comment]
Hmm, you have a bit of a mixture of conda-forge and defaults packages, but the relevant ones all seem to come from conda-forge. 

Can you test if you get the same error when making a clean new env with just `conda create -n test python=3.6 geopandas -c conda-forge` ? In that case, it would be a packaging problem.

BTW, I recommend to not install `cartopy` with pip but also with conda, as that contains some binaries linked to geos, just as shapely, and not having both from conda can give problems.

**Comment by aborruso** [comment]
@jorisvandenbossche I have the same error with this simple binder conf https://github.com/aborruso/geopandashub/tree/master/binder

I have tried to install `libjson-c-dev` and `libjson-c` after I have seen the json dependency error.

Thank you again.

**Comment by jorisvandenbossche** [comment]
@aborruso OK, that's interesting, thanks for making that minimal binder example. That *should* work.

cc @ocefpaf binder is failing with only this environment.yaml file: https://github.com/aborruso/geopandashub/blob/master/binder/environment.yml 
Currently making a clean env with just geopandas locally (on linux) is not failing for me.


@aborruso you can check https://github.com/jorisvandenbossche/geopandas-tutorial/blob/master/environment.yml that I recently used for a tutorial, that is working on binder, so you can add "defaults" channel for now to get a working binder.

**Comment by aborruso** [comment]
Now it works https://github.com/aborruso/bashnotebook/tree/master/binder

**Comment by jorisvandenbossche** [comment]
@aborruso What did you change?

Anyhow, the minimal binder example is still failing? (that is something we should fix)

**Comment by jorisvandenbossche** [comment]
> @aborruso What did you change?

Ah, I see, adding defaults as I said. 
BTW, I still think you don't need eg `gdal-bin` and `spatialite-bin` in the apt.txt file

**Comment by aborruso** [comment]
@jorisvandenbossche the most important thing is from

```
channels:
  - conda-forge
  - defaults
```

to

```
channels:
  - defaults
  - conda-forge  
```

And the minimal binder workd with this order.

Whena I have had the error I have `conda-forge` and than `defaults`

**Comment by jorisvandenbossche** [comment]
OK, opened an issue on the fiona feedstock: https://github.com/conda-forge/fiona-feedstock/issues/76

**Comment by ocefpaf** [comment]
This is ultimately an issue with `gdal` getting built with an older, and unpinned, `json-c`. I removed the new `json-c` as a workaround and we'll fix the pinning soon.

Sorry for that.

PS: Ideally try to open conda-forge issues in out feedstocks rather than the upstream source b/c they are 99% of the time related to the packaging and not the Software.

**Comment by matteodefelice** [comment]
I had the same problem and I fixed it by installing a specific build of `libgdal`. In fact I realised that creating a new clean environment as suggested by @jorisvandenbossche the build of `libgdal` was different from my current one.
This fixed everything: `conda install libgdal=2.2.4=ha6e4b24_6`

**Comment by helmogey** [comment]
i had the same problem and worked by using " pip3 install geopandas "
